examples/x86_64.swiftinterface

```diff
@@ -0,0 +1,507 @@
+  public var message: Swift.String
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ChatMessageEditedEvent : AzureCommunicationChat.BaseChatMessageEvent {
+  public var message: Swift.String
+  public var editedOn: AzureCore.Iso8601Date?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ChatMessageDeletedEvent : AzureCommunicationChat.BaseChatMessageEvent {
+  public var deletedOn: AzureCore.Iso8601Date?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class TypingIndicatorReceivedEvent : AzureCommunicationChat.BaseChatEvent {
+  public var version: Swift.String
+  public var receivedOn: AzureCore.Iso8601Date?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ReadReceiptReceivedEvent : AzureCommunicationChat.BaseChatEvent {
+  public var chatMessageId: Swift.String
+  public var readOn: AzureCore.Iso8601Date?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ChatThreadCreatedEvent : AzureCommunicationChat.BaseChatThreadEvent {
+  public var createdOn: AzureCore.Iso8601Date?
+  public var properties: AzureCommunicationChat.SignalingChatThreadProperties?
+  public var participants: [AzureCommunicationChat.SignalingChatParticipant]?
+  public var createdBy: AzureCommunicationChat.SignalingChatParticipant?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ChatThreadPropertiesUpdatedEvent : AzureCommunicationChat.BaseChatThreadEvent {
+  public var properties: AzureCommunicationChat.SignalingChatThreadProperties?
+  public var updatedOn: AzureCore.Iso8601Date?
+  public var updatedBy: AzureCommunicationChat.SignalingChatParticipant?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ChatThreadDeletedEvent : AzureCommunicationChat.BaseChatThreadEvent {
+  public var deletedOn: AzureCore.Iso8601Date?
+  public var deletedBy: AzureCommunicationChat.SignalingChatParticipant?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ParticipantsAddedEvent : AzureCommunicationChat.BaseChatThreadEvent {
+  public var addedOn: AzureCore.Iso8601Date?
+  public var participantsAdded: [AzureCommunicationChat.SignalingChatParticipant]?
+  public var addedBy: AzureCommunicationChat.SignalingChatParticipant?
+  @objc deinit
+}
+@_hasMissingDesignatedInitializers public class ParticipantsRemovedEvent : AzureCommunicationChat.BaseChatThreadEvent {
+  public var removedOn: AzureCore.Iso8601Date?
+  public var participantsRemoved: [AzureCommunicationChat.SignalingChatParticipant]?
+  public var removedBy: AzureCommunicationChat.SignalingChatParticipant?
+  @objc deinit
+}
+public enum ChatEventId : Swift.String {
+  case chatMessageReceived
+  case typingIndicatorReceived
+  case readReceiptReceived
+  case chatMessageEdited
+  case chatMessageDeleted
+  case chatThreadCreated
+  case chatThreadPropertiesUpdated
+  case chatThreadDeleted
+  case participantsAdded
+  case participantsRemoved
+  public init?(rawValue: Swift.String)
```

Comment:
Should this init throw instead of just being optional? I assume it is just optional in the case if a string comes in that we don't recognize?

Reply:
I am not actually sure why swiftinterface is generated this way - it does throw if we get an unrecognized event id
https://github.com/Azure/azure-sdk-for-ios/blob/6e984283fa291312a12ed561a61b1b69734bcc50/sdk/communication/AzureCommunicationChat/Source/Signaling/Events/ChatEvent.swift#L900

Reply:
Oh interesting. I wonder if it is because it is an enum.